libs/common/src/lib/prompt-lab/prompt-lab.component.ts

```diff
@@ -0,0 +1,308 @@
+                            return of(null);
+                          }),
+                        ),
+                    ),
+                  ),
+                ),
+              );
+            }),
+          ),
+          // Then we put back the KB as it was
+          this.sdk.currentKb.pipe(
+            take(1),
+            concatMap((kb) => {
+              const modelBackup = this.configBackup?.[GENERATIVE_MODEL_KEY];
+              return kb.setConfiguration({ [GENERATIVE_MODEL_KEY]: modelBackup });
+            }),
+            tap(() => {
+              this.hasResults.next(this.results.length > 0);
+              this.loadingModal?.close();
+            }),
+          ),
+        ),
+      ),
+    );
+  }
+
+  private _addResult(
+    entry: {
+      model: string;
+      query: string;
+      prompt: string | undefined;
+      result: Chat.Answer | IErrorResponse | string;
+    },
+    requestCount: number,
+  ) {
+    const { model, query, prompt, result } = entry;
+    const queryEntry = this.results.find((entry) => entry.query === query);
+    const answer =
+      typeof result === 'string'
+        ? result
+        : result.type === 'answer'
+          ? (result as Chat.Answer).text
+          : `Error: ${result.detail}`;
+    if (queryEntry) {
+      const promptEntry = queryEntry.data.find((entry) => entry.prompt === prompt);
+      if (promptEntry) {
+        promptEntry.results.push({ model, answer });
+      } else {
+        queryEntry.data.push({ prompt, results: [{ model, answer }] });
+      }
+    } else {
+      this.results.push({ query, data: [{ prompt, results: [{ model, answer }] }] });
+    }
+
+    this.progress$.next((this.progress$.value || 0) + Math.ceil(100 / requestCount));
+  }
+
+  collapseAnswer(query: string, prompt: string | undefined, model: string) {
+    const fullId = query + prompt + model;
+    this.modelCollapsed[fullId] = !this.modelCollapsed[fullId];
+  }
+
+  downloadCsv() {
+    if (this.results.length === 0) {
+      return;
+    }
+    let csv = `Query,${this.selectedModels.map((model) => this.generativeModelPipe.transform(model)).join(',')}`;
+    this.results.forEach((item) => {
+      item.data.forEach(({ prompt, results }) => {
+        const answers = this.selectedModels
+          .map((model) => {
+            const modelResult = results.find((result) => result.model === model);
+            return `"${this.formatCellValue(modelResult?.answer || 'No answer')}"`;
+          })
+          .join(',');
+        csv += `\n"Query: ${this.formatCellValue(item.query)}${
+          prompt ? '\nPrompt: ' + this.formatCellValue(prompt) : ''
+        }",${answers}`;
+      });
+    });
+    const filename = `${new Date().toISOString().split('T')[0]}_Nuclia_models_comparison.csv`;
+    const blob = new Blob([csv], { type: 'text/csv;charset=utf-8' });
+    const link = document.createElement('a');
+    if (link.download !== undefined) {
+      const url = URL.createObjectURL(blob);
```

Comment:
The mdn docs suggests to clean this url `revokeObjectURL` once it's not in use. Would it be necessary in this case ?